java/kafka/consumer/src/main/java/KafkaConsumerExample.java

```diff
@@ -30,11 +45,21 @@ public static void main(String[] args) {
         Properties props = KafkaConsumerConfig.createProperties(config);
         int receivedMsgs = 0;
 
-        if (System.getenv("JAEGER_SERVICE_NAME") != null)   {
-            Tracer tracer = Configuration.fromEnv().getTracer();
-            GlobalTracer.registerIfAbsent(tracer);
+        TracingSystem tracingSystem;
+        tracingSystem = TracingSystem.forValue(config.getTracingSystem());
+
+        if (tracingSystem != null) {
+            if (tracingSystem == KafkaConsumerExample.TracingSystem.JAEGER) {
+                Tracer tracer = Configuration.fromEnv().getTracer();
+                GlobalTracer.registerIfAbsent(tracer);
 
-            props.put(ConsumerConfig.INTERCEPTOR_CLASSES_CONFIG, TracingConsumerInterceptor.class.getName());
+                props.put(ConsumerConfig.INTERCEPTOR_CLASSES_CONFIG, io.opentracing.contrib.kafka.TracingConsumerInterceptor.class.getName());
+            } else if (tracingSystem == KafkaConsumerExample.TracingSystem.OPENTELEMETRY) {
```

Comment:
I think you should be able to drop the `KafkaConsumerExample.` prefix here

Reply:
Dropped without any issues.